plugin/discovery/version_set.go

```diff
@@ -0,0 +1,58 @@
+package discovery
+
+import (
+	version "github.com/hashicorp/go-version"
+)
+
+// A ConstraintsStr is a string containing a possibly-invalid representation
+// of a version constraint provided in configuration. Call Parse on it to
+// obtain a real Constraint object, or discover that it is invalid.
+type ConstraintsStr string
```

Comment:
The `Str` suffix here is trying to say "string representation provided by the user, possibly invalid". As you noticed, this is following the same idea as `VersionStr`, the string representation of a `Version`.

The naming of it as plural came from the fact that it's used to produce a `version.Constraints` object, but I suppose that reason is vestigial now that I wrapped that up in a `VersionSet` anyway. So agreed, I will change it to `ConstraintStr` since that feels more natural.